### DISCUSSION TITLE
auto common

### DISCUSSION THREAD

**Discussion by amitu** [fastn-stack/fastn#1585] [Ideas & RFCs]
What if we allow all common properties automatically on all user defined components, and automatically default pass them to the top level child of the component. If you want to pass it to some other child, or do something else with it, you define a field with same name and type, and pass it to whatever you want. If you define the any field you will have to manually pass it to whoever you want to pass.

**Comment by Arpita-Jaiswal** [comment]
This could be an interesting thing.
